plugins/firebase.js

```diff
@@ -1,16 +1,35 @@
-import Firebase from 'firebase/app'
+import firebase from 'firebase/app'
 import 'firebase/auth'
 import 'firebase/firestore'
 import 'firebase/analytics'
 
-export default ({ env: { firebase } }) => {
-  if (Firebase.apps.length > 0) {
-    return
+let track = function(...params) {
+  if (process.env.FIREBASE_ANALYTICS_DEBUG) {
+    console.log('[track]', ...params)
   }
+}
+
+let analytics
+
+let googleApiKey
+
+if (!firebase.apps.length) {
+  const configString = process.env.FIREBASE_CONFIG
+  if (!configString) {
+    throw new Error('FIREBASE_CONFIG is not defined')
+  }
+
+  const config = JSON.parse(configString)
+  googleApiKey = config.apiKey
```

Comment:
seeing this, `googleApiKey` is from config env.

Reply:
Exactly